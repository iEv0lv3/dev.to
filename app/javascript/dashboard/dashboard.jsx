import { h, Component } from 'preact';
import displayRead from '../../assets/images/display_read.png';
import displayClick from '../../assets/images/display_click.png';
import displayTime from '../../assets/images/display_time.png';
import displayComment from '../../assets/images/display_comment.png';


export class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    /* eslint-disable */ const { visits, visit, comments, time, articles, avgArticles, words, avgWords } = this.props;
console.log(avgArticles)
    return (
      <div className="display-div">
        <div className="user-visits stats">
          <div className="title">VISITS</div>
          <img className="icons" src={displayClick}></img>
          {visits} total
          {visit} avg/day
        </div>
        <div className="comments stats">
          <div className="title">COMMENTS</div>
          <img className="icons" src={displayComment}></img>
          {comments} total
        </div>
        <div className="time stats">
          <div className="title">TIME SPENT</div>
          <img className="icons" src={displayTime}></img>
          {time} per day
        </div>
        <div className="articles">
          <div className="border">
            ARTICLES READ
            <img className="icons article-icon" src={displayRead}></img>
            {articles} total
            {avgArticles} avg/day
          </div>
            WORDS READ
            {words} total
            {avgWords} avg/day
        </div>
      </div>
    );
  }
}
