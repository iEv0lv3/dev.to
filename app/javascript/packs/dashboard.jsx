import { h, render } from 'preact';
import { Dashboard } from '../dashboard/dashboard';

function loadElement() {
  const root = document.getElementById('dashboard');
  console.log(root.dataset.avgArticles)
  if (root) {
    render(
      <Dashboard
        visits={root.dataset.visits}
        visit={root.dataset.visit}
        comments={root.dataset.comments}
        time={root.dataset.time}
        articles={root.dataset.articles}
        avgArticles={root.dataset.avgarticles}
        words={root.dataset.words}
        avgWords={root.dataset.avgwords}
      />,
      root,
      root.firstElementChild,
    );
  }
}

window.InstantClick.on('change', () => {
  loadElement();
});

loadElement();
