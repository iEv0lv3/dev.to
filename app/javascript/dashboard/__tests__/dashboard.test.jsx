import { h } from 'preact';
import render from 'preact-render-to-json';
import { Dashboard } from '../dashboard';

describe('<Dashboard />', () => {
  it('renders properly', () => {
    const tree = render(
      <Dashboard
        visits="1"
        visit="2"
        comments="3"
        time="4"
        articles="5"
        avgArticles="6"
        words="7"
        avgWords="8"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
