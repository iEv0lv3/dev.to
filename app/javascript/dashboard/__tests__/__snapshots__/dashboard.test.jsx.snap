// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders properly 1`] = `
<div
  class="display-div"
>
  <div
    class="user-visits stats"
  >
    <div
      class="title"
    >
      VISITS
    </div>
    <img
      class="icons"
      src=""
    />
    1 total2 avg/day
  </div>
  <div
    class="comments stats"
  >
    <div
      class="title"
    >
      COMMENTS
    </div>
    <img
      class="icons"
      src=""
    />
    3 total
  </div>
  <div
    class="time stats"
  >
    <div
      class="title"
    >
      TIME SPENT
    </div>
    <img
      class="icons"
      src=""
    />
    4 per day
  </div>
  <div
    class="articles"
  >
    <div
      class="border"
    >
      ARTICLES READ
      <img
        class="icons article-icon"
        src=""
      />
      5 total6 avg/day
    </div>
    WORDS READ7 total8 avg/day
  </div>
</div>
`;
